### ISSUE TITLE
[java] GuardLogStatement only detects guard methods immediately around it.

### ISSUE THREAD

**Issue by Frettman** [pmd/pmd#4282]
**Affects PMD Version:** 6.53.0

**Rule:** [GuardLogStatement](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:**

I have an additional `if` between the guard method and the actual call to a log method (see example below). In the `else` I'm directly calling a (simple getter) method to provide an argument to the log method, which triggers the need for a guard. This raises a violation: `Logger calls should be surrounded by log level guards.`
The rule does not recognize that there's a guard around it, just not as direct parent in the syntax tree.

**Code Sample demonstrating the issue:**

```java
if (LOG.isWarnEnabled())
{
  if (someCondition)
  {
    String msg = "Message 1 with one argument: {}.";
    ... // do some more stuff with the message
    LOG.warn(msg, arg1);
  }
  else
  {
    String msg = "Message 2 with two arguments: {}, {}.";
    ... // do some more stuff with the message
    LOG.warn(msg, arg1, someObject.someMethod());
  }
}
```

**Expected outcome:**

PMD should look further around it to detect a guard method.

**Running PMD through:** Maven | Eclipse Plug-in

**Comment by SheikhArhaan** [comment]
facing the same issue

**Comment by krallus** [comment]
Interestingly, this issue was did not occur for me in PMD 7.2, 7.3, or 7.4.  However, it is a problem in PMD versions 7.5 and 7.6.
